dashboard/utils_test.py

```diff
@@ -363,9 +367,108 @@ def test_is_enrolled_mmtrack_fa(self):
         )
         assert mmtrack.is_enrolled_mmtrack(course_id) is True
 
-    def test_has_passed_course_normal(self):
+    @override_settings(FEATURES={"FINAL_GRADE_ALGO": "v1"})
+    @patch('dashboard.utils.MMTrack.extract_final_grade', autospec=True)
+    def test_has_passed_course(self, extr_grade_mock):
+        """
+        Test for has_passed_course method.
+        V1 new behavior soon to be default
+        """
+        course_id = "course-v1:odl+FOO101+CR-FALL15"
+        grade_mock = MagicMock(passed=True)
+        extr_grade_mock.return_value = grade_mock
+
+        mmtrack = MMTrack(
+            user=self.user,
+            program=self.program,
+            edx_user_data=self.cached_edx_user_data
+        )
+
+        assert mmtrack.has_passed_course(course_id) is True
+        extr_grade_mock.assert_called_once_with(mmtrack, course_id)
+
+        extr_grade_mock.reset_mock()
```

Comment:
If you have to call `reset_mock()`, you're trying to test too much in one test function. Please split this into two different tests.